sdk/physical/cache.go

```diff
@@ -274,7 +283,7 @@ func (c *cache) cloneWithStorage(b Backend) *cache {
 	// fresh, localized cache. This is globally sub-optimal (as it starts
 	// with an empty cache), but easiest to implement (as the transaction can
 	// modify its cache as it pleases).
-	cacheCopy := NewCache(b, c.size, c.logger, c.metricSink).(*cache)
+	cacheCopy := newCache(b, c.size/TransactionCacheFactor, c.logger, c.metricSink).(*cache)
```

Comment:
I suppose there is no risk to have a division by zero here.

Reply:
@DanGhita no, the numerator is non-zero by guard in NewCache and the denominator is a packages-level constant defined above (set to the value of another packages-level constant which is also non-zero).